### PR TITLE
Feature/Metrics

### PR DESCRIPTION
**Pull request by iGxnon**
Base on #10 

Please briefly answer these questions:

Add metrics exporter to prometheus.

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    Operators need to expose some metrics for external collection, and Prometheus is an excellent option. This PR exposes two metrics:

    1. Reconcile Duration Histogram
    2. Reconcile Failed Count

    These two metrics are sufficient for the current system. Afterward, when we add a sidecar, we will include sidecar-related metrics, including but not limited to:

    - Evaluate & Reconcile Duration Histogram
    - Evaluate & Reconcile Failed Count
    - Backup Duration Histogram
    - Backup Failed Count 

* what changes does this pull request make?

    As above

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

    No

### CODE REVIEW

File: operator-k8s/src/router.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    let mut buf2 = Vec::new();
+    if let Err(err) = encoder.encode(&prometheus::gather(), &mut buf2) {
+        error!("failed to encode prometheus metrics: {}", err);
+        return String::new();
+    }
+    res.push_str(&String::from_utf8_lossy(&buf2));
+    res
+}
+
+/// healthz handler
```
Comment (Phoenix500526) [inline operator-k8s/src/router.rs:27]: typo, it's "healthy"

Comment (iGxnon) [reply operator-k8s/src/router.rs:27]: I guess it is not a typo.
FYI: https://stackoverflow.com/questions/43380939/where-does-the-convention-of-using-healthz-for-application-health-checks-come-f

File: operator-k8s/src/controller/cluster/mod.rs [hunk 2]
```diff
@@ -1,7 +1,81 @@
...
+impl Metrics for ClusterMetrics {
+    /// Register metrics
+    fn register(&self, registry: &Registry) -> Result<(), Error> {
+        registry.register(Box::new(self.reconcile_duration.clone()))?;
+        registry.register(Box::new(self.reconcile_failed_count.clone()))
+    }
+
+    /// Record duration
+    fn record_duration(&self) -> HistogramTimer {
+        self.reconcile_duration.start_timer()
```
Comment (bsbds) [inline operator-k8s/src/controller/cluster/mod.rs:40]: Does this has a `stop timer` method that need to be called?

Comment (iGxnon) [reply operator-k8s/src/controller/cluster/mod.rs:40]: `stop timer` will be invoked when HistogramTimer dropped.

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by bsbds — APPROVED** [review]

**Pull request merged**